extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -158,19 +164,21 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
       if (elapsedTime < DURATION_MS) {
         const easing = easeInOutQuad(elapsedTime / DURATION_MS);
         this.storeService_.dispatch(Action.SET_PANNING_MEDIA_STATE, {
-          x: startPos.x + (this.x_ - startPos.x) * easing,
-          y: startPos.y + (this.y_ - startPos.y) * easing,
-          zoom: startPos.zoom + (this.zoom_ - startPos.zoom) * easing,
+          ...getPanningMediaState(),
+          [this.groupId_]: {
+            x: startPos.x + (this.x_ - startPos.x) * easing,
+            y: startPos.y + (this.y_ - startPos.y) * easing,
+            zoom: startPos.zoom + (this.zoom_ - startPos.zoom) * easing,
+          },
```

Comment:
You might be able to move this to the `action()` method of `amp-story-store-service.js` so you don't need to fetch the state so many times here and below. You could dispatch `{group: "group-1", data: {x, y, zoom}}` and use the group and data to update the state accordingly, and it could save some calls to `getPanningMediaState`

Reply:
This is awesome. Thank you for providing a detailed sample implementation like this. It helps me understand the full concept more clearly and gives me strong footing to build off of.

I see how this idea is to merge the update on the store level instead of component level.

Do you think there is a benefit to having an additional level with the `group` and `data` keys? 
Otherwise we could do:
`const updatedState = {...state[StateProperty.PANNING_MEDIA_STATE],...data};` 
and keep the map shallow.

Reply:
That also works! If I understand you'd still need to key the data with `[data.group]: data`